libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -284,45 +244,37 @@ void UARTDriver::set_blocking_writes(bool blocking)
  */
 bool UARTDriver::tx_pending()
 {
-    return !BUF_EMPTY(_writebuf);
+    return (_writebuf.available() > 0);
 }
 
 /*
   return the number of bytes available to be read
  */
 uint32_t UARTDriver::available()
 {
-    if (!_initialised) {
-        return 0;
-    }
-    uint16_t _tail;
-    return BUF_AVAILABLE(_readbuf);
+    return _readbuf.available();
```

Comment:
Should the check for *_initialised* really be removed? I mean, it can happen that *_writebuf* was set to the correct size and the *_readbuf* wasn't or vice-versa.